src/main/java/org/minions/devfund/noemiguzman/Squarelotron.java

```diff
@@ -0,0 +1,297 @@
+package org.minions.devfund.noemiguzman;
+
+/**
+ * Created by Noemi on 18-07-18.
+ */
+public class Squarelotron {
+
+    /*
+     * instance variables
+     */
+    private int[][] squarelotron;
+    private int size;
+    private static final int MAX_SIZE = 99;
+    private static final int NUM_SIDES = 4;
+    private static final int DIMENSION = 2;
+
+    /**
+     *
+     * @return size of squarelotron
+     */
+
+    public int getSize() {
+        return size;
+    }
+
+    /**
+     *
+     * @return current squarelotron
+     */
+    public int[][] getSquarelotron() {
+        return squarelotron.clone();
+    }
+    /**
+     * constructor It fills the 2-dimensional array with the numbers 1 to n squared.
+     *
+     * @param n - the dimension of the squarelotron
+     */
+    public Squarelotron(int n) {
+        this.squarelotron = new int[n][n];
+        size = n;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                this.squarelotron[i][j] = i * n + j + 1;
+            }
+        }
+    }
+
+    /**
+     * additional constructor.
+     *
+     * @param array - 1d array convert to a squarelotron
+     */
+    public Squarelotron(final int[] array) {
+        Squarelotron newSquare = Squarelotron.makeSquarelotron(array);
+        this.squarelotron = new int[newSquare.size][newSquare.size];
+        this.size = newSquare.size;
+
+        for (int i = 0; i < newSquare.size; i++) {
+            for (int j = 0; j < newSquare.size; j++) {
+                this.squarelotron[i][j] = newSquare.squarelotron[i][j];
+            }
+        }
+    }
+
+    /**
+     * method to construct a Squarelotron using an array.
+     *
+     * @param array - pass in array
+     * @return - a new Squarelotron class
+     */
+    public static Squarelotron makeSquarelotron(final int[] array) {
+        // test square length
+        int sqrt = (int) Math.sqrt(array.length);
+        if (sqrt * sqrt != array.length) {
+            throw new IllegalArgumentException("bad array provided");
+        }
+        // test the length is between 0 to MAX_SIZE
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] < 0 || array[i] > MAX_SIZE || array[i] != array[i]) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Correct one of the identical sub-expressions on both sides of operator "!=" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1764)